### DISCUSSION TITLE
ModuleNotFoundError: No module named 'thinc.types'

### DISCUSSION THREAD

**Discussion by jacekwachowiak** [explosion/spaCy#7910] [Help: Installation] [answered]
#7219 
Hi, I am having problems with importing Spacy, I have used a requirements file to install all libraries to a new venv and what worked on an older one does not work anymore, I get the error:

```---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-1-76a01d9c502b> in <module>
----> 1 import spacy

C:\Users\XXX\AppData\Roaming\Python\Python37\site-packages\spacy\__init__.py in <module>
      8 
      9 # These are imported as part of the API
---> 10 from thinc.api import prefer_gpu, require_gpu, require_cpu  # noqa: F401
     11 from thinc.api import Config
     12 

C:\Users\XXX\AppData\Roaming\Python\Python37\site-packages\thinc\__init__.py in <module>
      3 
      4 from .about import __version__
----> 5 from .config import registry

C:\Users\XXX\AppData\Roaming\Python\Python37\site-packages\thinc\config.py in <module>
     20 import re
     21 
---> 22 from .types import Decorator
     23 
     24 

ModuleNotFoundError: No module named 'thinc.types'
```

My settings:

```$ pip list | grep spacy
spacy               3.0.0
spacy-alignments    0.8.3
spacy-legacy        3.0.5
$ pip list | grep thinc
thinc               8.0.3
```

## Info about spaCy

- **spaCy version:** 3.0.0
- **Platform:** Windows-10-10.0.18362-SP0
- **Python version:** 3.7.4
- **Pipelines:** de_core_news_lg (3.0.0)

I can't find any solution and the other closed ticket is about an old version, which is not my issue. I tried to reinstall everything, with or without transformers but to no avail.

**Comment by polm** [comment]
Sorry you're having trouble with this. 8.0.3 is the most recent release of thinc and it definitely has a `thinc.types` that you should be able to import. 

1. If you open a python prompt from the terminal and `import thinc.types` does it work?
2. Can you create a new virtualenv and install spaCy from scratch, not from your requirements.txt? (not clear from your comment if that's what you meant by reinstall)
3. Can you post the whole output of `pip freeze` for us to check?

**Reply by jacekwachowiak** [reply]
1. No, I get the ```ModuleNotFoundError: No module named 'thinc.types'``` (in Jupyter)
2. I did and it works :) so in a way it's fixed (but would mean one of the other libraries had a conflict?)
3. List from my pip freeze from requirements.txt I used before (it has some other libraries I needed)
```
$ pip freeze
argon2-cffi==20.1.0
async-generator==1.10
attrs==20.3.0
backcall==0.2.0
backports.csv==1.0.7
beautifulsoup4==4.9.3
bleach==3.3.0
blis==0.7.4
cassandra-driver==3.20.0
catalogue==2.0.3
certifi==2020.12.5
cffi==1.14.5
chardet==4.0.0
cheroot==8.5.2
CherryPy==18.6.0
click==7.1.2
colorama==0.4.4
cql==1.4.0
cqlsh==5.0.4
cryptography==3.4.7
cycler==0.10.0
cymem==2.0.5
Cython==0.29.14
de-core-news-lg==3.0.0
decorator==5.0.7
defusedxml==0.7.1
elasticsearch==7.0.5
entrypoints==0.3
feedparser==6.0.2
filelock==3.0.12
future==0.18.2
gensim==3.8.1
germalemma==0.1.1
idna==2.10
importlib-metadata==4.0.1
ipykernel==5.5.3
ipython==7.22.0
ipython-genutils==0.2.0
ipywidgets==7.6.3
jaraco.classes==3.2.1
jaraco.collections==3.3.0
jaraco.functools==3.3.0
jaraco.text==3.5.0
jedi==0.18.0
Jinja2==2.11.3
joblib==1.0.1
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client==6.1.12
jupyter-console==6.4.0
jupyter-core==4.7.1
jupyterlab-pygments==0.1.2
jupyterlab-widgets==1.0.0
kiwisolver==1.3.1
langdetect==1.0.7
lxml==4.6.3
MarkupSafe==1.1.1
matplotlib==2.2.4
mistune==0.8.4
more-itertools==8.7.0
murmurhash==1.0.5
mysqlclient==2.0.3
nbclient==0.5.3
nbconvert==6.0.7
nbformat==5.1.3
nest-asyncio==1.5.1
nltk==3.6.2
notebook==6.3.0
numpy==1.20.2
packaging==20.9
pandas==1.2.4
pandocfilters==1.4.3
parso==0.8.2
pathy==0.5.2
Pattern==3.6
pdfminer.six==20201018
pickleshare==0.7.5
portend==2.7.1
preshed==3.0.5
prometheus-client==0.10.1
prompt-toolkit==3.0.18
pycparser==2.20
pydantic==1.7.3
Pygments==2.8.1
pyparsing==2.4.7
Pyphen==0.10.0
pyrsistent==0.17.3
python-crfsuite==0.9.7
python-dateutil==2.8.1
python-docx==0.8.10
pytz==2021.1
pywin32==300
pywinpty==0.5.7
pyzmq==22.0.3
qtconsole==5.0.3
QtPy==1.9.0
regex==2021.4.4
requests==2.25.1
sacremoses==0.0.45
scikit-learn==0.24.1
scipy==1.6.3
seaborn==0.11.1
Send2Trash==1.5.0
sgmllib3k==1.0.0
six==1.15.0
sklearn-crfsuite==0.3.6
smart-open==3.0.0
sortedcontainers==2.3.0
soupsieve==2.2.1
spacy==3.0.0
spacy-alignments==0.8.3
spacy-legacy==3.0.5
srsly==2.4.1
stop-words==2018.7.23
tabulate==0.8.9
tempora==4.0.2
terminado==0.9.4
testpath==0.4.4
textvec==2.0
thinc==8.0.3
threadpoolctl==2.1.0
thrift==0.13.0
tokenizers==0.10.2
torch==1.8.1
tornado==6.1
tqdm==4.60.0
traitlets==5.0.5
transformers==4.5.1
typer==0.3.2
typing-extensions==3.7.4.3
urllib3==1.26.4
wasabi==0.8.2
wcwidth==0.2.5
webencodings==0.5.1
widgetsnbextension==3.5.1
zc.lockfile==2.0
zipp==3.4.1
```

**Comment by svlandeg** [comment, marked as answer]
> No, I get the ModuleNotFoundError: No module named 'thinc.types' (in Jupyter)

Could you double check this in a command prompt, outside of Jupyter? The list of requirements you've cited here looks correct, so we need to figure out whether it's an issue with your environment itself, or with the combination of using this env in Jupyter. We've seen issues before where Jupyter was either not started in the correct environment, or was for some reason pulling in installations from the system outside of the environment.

**Reply by jacekwachowiak** [reply]
Yes, you're right! It's Jupyter's problem - even though the kernel is obviously restarted every time, it somehow crashes on spacy and thinc in an old notebook - I have started an empty one and I can import it even from the old venv.
